### PR TITLE
Feature: Empty selected file or directory

### PR DESCRIPTION
**Pull request by zhangsean**
PR for #62.

### CODE REVIEW

File: tui/actions.go [hunk 1]
```diff
@@ -112,30 +113,56 @@ func (ui *UI) AnalyzePath(path string, parentDir *analyze.Dir) error {
 	return nil
 }
 
-func (ui *UI) deleteSelected() {
+func (ui *UI) deleteSelected(isEmpty bool) {
```
Comment (dundee) [inline tui/actions.go:116]: `isEmpty` sounds like a flag indicating that the target directory is empty. Maybe `shouldEmpty` would be better? What do you think?

File: tui/actions.go [hunk 2]
```diff
@@ -112,30 +113,56 @@ func (ui *UI) AnalyzePath(path string, parentDir *analyze.Dir) error {
...
+		acting = "emptying"
+	} else {
+		action = "delete "
+		acting = "deleting"
+	}
+	modal := tview.NewModal().SetText(strings.Title(acting) + " " + selectedItem.GetName() + "...")
+	ui.pages.AddPage(acting, modal, true, true)
+
+	var currentDir *analyze.Dir
+	var currentItems []analyze.Item
```
Comment (dundee) [inline tui/actions.go:132]: What is the meaning of `current` here? Wouldn't `itemsToTruncate` be better?

File: tui/actions.go [hunk 3]
```diff
@@ -112,30 +113,56 @@ func (ui *UI) AnalyzePath(path string, parentDir *analyze.Dir) error {
...
-			msg := "Can't delete " + selectedFile.GetName()
-			ui.app.QueueUpdateDraw(func() {
-				ui.pages.RemovePage("deleting")
-				ui.showErr(msg, err)
-			})
-			if ui.done != nil {
-				ui.done <- struct{}{}
+		for _, item := range currentItems {
+			var err error
+			if isEmpty && !selectedItem.IsDir() {
```
Comment (dundee) [inline tui/actions.go:146]: I think you could do this check just once outside the loop and store the function (emptier or remover) into a variable to be executed here.

**Review by dundee — CHANGES_REQUESTED** [review]
Very nice patch!

I have just some small ideas for improvement.

Also tests have to be fixed (I can help you with that, there is small help function to show the content of the simulation buffer) and some added to cover the new funcionality.

File: tui/actions.go (hunk 1 above)
Comment (zhangsean) [reply tui/actions.go:116]: Yes, `shouldEmpty` is much better.

File: tui/actions.go (hunk 2 above)
Comment (zhangsean) [reply tui/actions.go:132]: Basically, `current` means `ui.current`. `itemsToTruncate` or `itemsToDelete` would be better.

File: tui/actions.go (hunk 3 above)
Comment (zhangsean) [reply tui/actions.go:146]: Thanks for your excellent code suggestions.

**Comment by dundee** [comment]
Could you also fix the tests and add some more? Or should I do it?

**Comment by zhangsean** [comment]
Please help me for the tests.

**Pull request merged**